tests/test-constraintgraph.cc

```diff
@@ -141,14 +141,29 @@ BOOST_AUTO_TEST_CASE (Initialization)
   using hpp_test::graph_;
   using hpp::pinocchio::LiegroupElement;
   using hpp::pinocchio::LiegroupSpace;
+  using hpp::constraints::ComparisonTypes_t;
+  using hpp::constraints::ImplicitPtr_t;
+  using hpp::constraints::EqualToZero;
+  using hpp::constraints::Equality;
   using hpp::constraints::LockedJoint;
   using hpp::manipulation::graph::Edge;
   using hpp::manipulation::graph::EdgePtr_t;
 
   initialize(true);
   hpp::manipulation::DevicePtr_t robot(graph_->robot());
-  graph_->addNumericalConstraint(LockedJoint::create
-    (robot->jointAt(1), LiegroupElement(LiegroupSpace::R1(true))));
+  ImplicitPtr_t constraint(LockedJoint::create
+			   (robot->jointAt(1), LiegroupElement
+			    (LiegroupSpace::R1(true))));
+  graph_->addNumericalConstraint(constraint);
+  // Check that states refuse parameterizable constraints
+  constraint->comparisonType(ComparisonTypes_t(1,Equality));
+  bool exception(false);
+  try{
+    n1->addNumericalConstraint(constraint);
+  } catch(const std::logic_error& err){
+    exception = true;
+  }
+  BOOST_CHECK(exception);
```

Comment:
```suggestion
  BOOST_CHECK_THROW(n1->addNumericalConstraint(constraint), std::logic_error);
```

See https://www.boost.org/doc/libs/1_75_0/libs/test/doc/html/boost_test/utf_reference/testing_tool_ref/assertion_boost_level_throw.html